scmrepo/git/backend/base.py

```diff
@@ -240,7 +241,7 @@ def push_refspecs(
     def fetch_refspecs(
         self,
         url: str,
-        refspecs: Union[str, Iterable[str]],
+        refspecs: Union[bytes, List[bytes], List[Tuple[bytes, bytes]]],
```

Comment:
In scmrepo our methods should take strings, not bytes, and do any conversion to bytes in the dulwich backend.

Basically the existing typehint for this method is what we want, and the dulwich implementation should be doing the conversion to
```
[os.fsencode(refspec) for refspec in refspecs]
```
when we call the underlying dulwich method

Reply:
also just to be clear, the fix for this also needs to account for when the user passes scmrepo a single string